lib/bgs/dependents.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+module BGS
+  class Dependents
+    def initialize(proc_id:, payload:, user:)
+      @proc_id = proc_id
+      @payload = payload
+      @dependents = []
+      @dependents_application = @payload['dependents_application']
+      @user = user
+    end
+
+    # rubocop:disable Metrics/PerceivedComplexity
+    # rubocop:disable Metrics/CyclomaticComplexity
+    def create
+      report_674 if @payload['report674']
+      add_children if @payload['add_child']
+      report_deaths if @payload['report_death']
+      report_divorce if @payload['report_divorce']
+      report_stepchild if @payload['report_stepchild_not_in_household']
+      report_child_event('child_marriage') if @payload['report_marriage_of_child_under18']
+      report_child_event('not_attending_school') if @payload['report_child18_or_older_is_not_attending_school']
+
+      @dependents
+    end
+
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
+
+    private
+
+    def add_children
+      @dependents_application['children_to_add'].each do |child_info|
+        child = BGSDependents::Child.new(child_info)
+        formatted_info = child.format_info
+        participant = bgs_service.create_participant(@proc_id)
+
+        bgs_service.create_person(person_params(child, participant, formatted_info))
+        send_address(child, participant, child.address(@dependents_application))
+
+        @dependents << child.serialize_dependent_result(
+          participant,
+          'Child',
+          formatted_info['family_relationship_type'],
+          {
+            marriage_termination_type_code: formatted_info['reason_marriage_ended'],
+            type: 'child'
+          }
+        )
+      end
+    end
+
+    def report_deaths
+      @dependents_application['deaths'].each do |death_info|
+        death = BGSDependents::Death.new(death_info)
+        relationship_types = death.relationship_type(death_info)
+
+        next if relationship_types[:family] == 'Child' # BGS does not support child death at this time
+
+        formatted_info = death.format_info
+        death_info['location']['state_code'] = death_info['location'].delete('state')
+        participant = bgs_service.create_participant(@proc_id)
+        bgs_service.create_person(person_params(death, participant, formatted_info))
+        # I think we need the death_location instead of creating an address
+        # There is no support in the API for death location
+        # create_address(@proc_id, participant[:vnp_ptcpnt_id], death_info['location'])
```

Comment:
Can we remove this commented-out code, or will this be used in the future?

Reply:
I've modified this comment - removed the old/unused code but left a comment that we will need to add functionality there once BGS implements it.